### PR TITLE
Revert "Update opentelemetry dependencies to the latest release."

### PR DESCRIPTION
**Pull request by dashpole**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

Revert due to https://github.com/kubernetes/kubernetes/pull/125575#issuecomment-2191292714

Related to https://github.com/kubernetes/kubernetes/issues/125688

#### Special notes for your reviewer:

The PR could not be automatically reverted, so i've manually downgraded all dependencies updated by the PR

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by dashpole** [comment]
@liggitt @dims @thockin @serathius

File: go.mod [hunk 1]
```diff
@@ -79,7 +79,7 @@ require (
 	golang.org/x/time v0.3.0
 	golang.org/x/tools v0.21.0
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20240520151616-dc85e6b867a5
-	google.golang.org/grpc v1.64.0
+	google.golang.org/grpc v1.63.0
```
Comment (liggitt) [inline go.mod:82]: hrmm, this was 1.59.0 before

**Comment by dashpole** [comment]
Looks like some dependencies weren't downgraded when I pinned them.

File: go.mod [hunk 2]
```diff
@@ -62,12 +62,12 @@ require (
 	go.etcd.io/etcd/api/v3 v3.5.14
 	go.etcd.io/etcd/client/pkg/v3 v3.5.14
 	go.etcd.io/etcd/client/v3 v3.5.14
-	go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful v0.52.0
+	go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful v0.42.0
 	go.opentelemetry.io/otel v1.27.0
```
Comment (liggitt) [inline go.mod:66]: this was 1.20.0

File: go.mod [hunk 3]
```diff
@@ -62,12 +62,12 @@ require (
...
-	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.27.0
+	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.20.0
 	go.opentelemetry.io/otel/sdk v1.27.0
```
Comment (liggitt) [inline go.mod:68]: 1.20.0

File: go.mod [hunk 4]
```diff
@@ -62,12 +62,12 @@ require (
...
 	go.opentelemetry.io/otel/trace v1.27.0
```
Comment (liggitt) [inline go.mod:69]: 1.20.0

File: go.mod [hunk 5]
```diff
@@ -62,12 +62,12 @@ require (
...
-	go.opentelemetry.io/proto/otlp v1.3.1
+	go.opentelemetry.io/proto/otlp v1.2.0
```
Comment (liggitt) [inline go.mod:70]: 1.0.0

File: go.mod [hunk 6]
```diff
@@ -79,7 +79,7 @@ require (
 	golang.org/x/time v0.3.0
 	golang.org/x/tools v0.21.0
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20240520151616-dc85e6b867a5
```
Comment (liggitt) [inline go.mod:81]: v0.0.0-20230822172742-b8732ec3820d

File: go.mod [hunk 7]
```diff
@@ -79,7 +79,7 @@ require (
...
-	google.golang.org/grpc v1.64.0
+	google.golang.org/grpc v1.63.0
 	google.golang.org/protobuf v1.34.1
```
Comment (liggitt) [inline go.mod:83]: google.golang.org/protobuf v1.33.0

File: go.mod [hunk 8]
```diff
@@ -162,7 +162,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-middleware v1.3.0 // indirect
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0 // indirect
 	github.com/grpc-ecosystem/grpc-gateway v1.16.0 // indirect
-	github.com/grpc-ecosystem/grpc-gateway/v2 v2.20.0 // indirect
+	github.com/grpc-ecosystem/grpc-gateway/v2 v2.19.1 // indirect
```
Comment (liggitt) [inline go.mod:165]: github.com/grpc-ecosystem/grpc-gateway/v2 v2.16.0

File: go.mod [hunk 9]
```diff
@@ -203,17 +203,17 @@ require (
 	go.etcd.io/etcd/raft/v3 v3.5.13 // indirect
 	go.etcd.io/etcd/server/v3 v3.5.13 // indirect
 	go.opencensus.io v0.24.0 // indirect
-	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.52.0 // indirect
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.52.0 // indirect
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.46.0 // indirect
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.44.0 // indirect
 	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.27.0 // indirect
```
Comment (liggitt) [inline go.mod:208]: v1.20.0

File: go.mod [hunk 10]
```diff
@@ -203,17 +203,17 @@ require (
...
 	go.opentelemetry.io/otel/metric v1.27.0 // indirect
```
Comment (liggitt) [inline go.mod:209]: v1.20.0

File: go.mod [hunk 11]
```diff
@@ -203,17 +203,17 @@ require (
...
 	go.starlark.net v0.0.0-20230525235612-a134d8f9ddca // indirect
 	go.uber.org/multierr v1.11.0 // indirect
 	golang.org/x/exp v0.0.0-20230515195305-f3d0a9c9a5cc // indirect
 	golang.org/x/mod v0.17.0 // indirect
 	golang.org/x/text v0.15.0 // indirect
-	google.golang.org/genproto v0.0.0-20230822172742-b8732ec3820d // indirect
-	google.golang.org/genproto/googleapis/api v0.0.0-20240520151616-dc85e6b867a5 // indirect
+	google.golang.org/genproto v0.0.0-20240227224415-6ceb2ff114de // indirect
```
Comment (liggitt) [inline go.mod:215]: v0.0.0-20230822172742-b8732ec3820d

File: go.mod [hunk 12]
```diff
@@ -203,17 +203,17 @@ require (
...
+	google.golang.org/genproto/googleapis/api v0.0.0-20240227224415-6ceb2ff114de // indirect
```
Comment (liggitt) [inline go.mod:216]: v0.0.0-20230822172742-b8732ec3820d

**Review by liggitt — COMMENTED** [review]
sorry this is so tedious

**Comment by liggitt** [comment]
I think it might be easier to just resolve the conflict in the revert commit in the go.mod files, then run update-vendor.sh to regen the go.sum files

**Comment by dashpole** [comment]
> I think it might be easier to just resolve the conflict in the revert commit in the go.mod files, then run update-vendor.sh to regen the go.sum files

That seemed to work better.

**Comment by dashpole** [comment]
/hold cancel 
per https://kubernetes.slack.com/archives/CHGFYJVAN/p1719412006120339

File: go.mod
Comment (liggitt) [inline go.mod:1]: comparing to go.mod before the original PR:

```
git diff e77a40c73e443345dfafccc9d8b8086e6e7f8746 HEAD -- go.mod
diff --git a/go.mod b/go.mod
index 10c07414400..cbdd10cfebf 100644
--- a/go.mod
+++ b/go.mod
@@ -145,7 +145,7 @@ require (
        github.com/exponent-io/jsonpath v0.0.0-20151013193312-d6023ce2651d // indirect
        github.com/fatih/camelcase v1.0.0 // indirect
        github.com/felixge/httpsnoop v1.0.3 // indirect
-       github.com/fxamacker/cbor/v2 v2.7.0-beta // indirect
+       github.com/fxamacker/cbor/v2 v2.7.0 // indirect
        github.com/go-errors/errors v1.4.2 // indirect
        github.com/go-logr/stdr v1.2.2 // indirect
        github.com/go-logr/zapr v1.3.0 // indirect
```

lgtm, that change was from the intermediate / unrelated update to the cbor library

**Comment by dashpole** [comment]
Seems like this PR still hit https://github.com/kubernetes/kubernetes/issues/125688

**Comment by liggitt** [comment]
> Seems like this PR still hit #125688 ([test run link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/125731/pull-kubernetes-unit/1805969363530944512))

cc @wojtek-t @p0lyn0mial @serathius for visibility that even with the grpc change reverted, the TestWatchSemantics test is still flaking. Is there another revert in flight that would address that?

**Comment by wojtek-t** [comment]
> cc @wojtek-t @p0lyn0mial @serathius for visibility that even with the grpc change reverted, the TestWatchSemantics test is still flaking. Is there another revert in flight that would address that?

heh....
introduced in https://github.com/kubernetes/kubernetes/pull/125721 as an attempt to speed up tests there (to reduce the time and avoid test timeouts)
looking into the error there it's "dumb" - we're comparing strings instead of the actual integers...

@liggitt - let me try to open a PR to fix that, unless you want to revert it...

**Comment by liggitt** [comment]
> introduced in #125721 as an attempt to speed up tests there (to reduce the time and avoid test timeouts) looking into the error there it's "dumb" - we're comparing strings instead of the actual integers...
> 
> @liggitt - let me try to open a PR to fix that, unless you want to revert it...

I don't have time to chase it, feel free to revert or fixup

**Comment by wojtek-t** [comment]
> I don't have time to chase it, feel free to revert or fixup

Will open a fix in few minutes

**Comment by wojtek-t** [comment]
/retest

[flake being fixed in https://github.com/kubernetes/kubernetes/pull/125733 ]

**Comment by serathius** [comment]
Confirmed it's an issue with upgrading google.golang.org/grpc

**Comment by serathius** [comment]
It happens between v1.63.0 and v1.64.0. We could consider upgrading opentelemetry assuming we can stay on grpc v1.63.0.

**Comment by serathius** [comment]
https://github.com/grpc/grpc-go/releases/tag/v1.64.0

```
metadata/metadata: Implement the fmt.Stringer interface for pretty printing MD (https://github.com/grpc/grpc-go/pull/7137)
```

**Comment by serathius** [comment]
OMG they made String() non-deterministic 

https://github.com/grpc/grpc-go/blob/98e5deebae600b6cd50d2f41bcb5155f8471d53f/metadata/metadata.go#L93-L106

**Pull request merged**